### ISSUE TITLE
Introduce a helper `Field` class

### ISSUE THREAD

**Issue by ddundo** [mesh-adaptation/goalie#181]
Currently information about each field is scattered around in different dictionaries of `MeshSeq` and `TimePartition`: `fields`, `_fs`, `field_types`, etc. I think we should introduce a class `Field` that would hold all of these at the same place.

**Comment by jwallwork23** [comment]
Sounds good, thanks. Note that I think fields were originally associated with `TimePartition` in case we wanted to support having different timesteps for different fields.

**Comment by ddundo** [comment]
> Sounds good, thanks. Note that I think fields were originally associated with `TimePartition` in case we wanted to support having different timesteps for different fields.

Further to this comment and to our discussion on the last meeting on setting options per field... perhaps allowing different interpolation method per field would be useful too?

**Comment by jwallwork23** [comment]
Yes, I agree. Different interpolators are more suitable for certain fields.
